compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -803,7 +810,9 @@ object CheckUnused:
           case _:tpd.Block => Local
           case _ => Other
 
-      case class UnusedSymbol(pos: SrcPos, name: Name, warnType: WarnTypes)
+      enum UnusedSymbol:
+          case Symbol(tree: tpd.NamedDefTree, warnType: WarnTypes)
+          case ImportSelector(pos: SrcPos, name: Name)
```

Comment:
Do we really need to distinguish them ?
Maybe we could model it more like
```scala
case class UnusedSymbol(defnTree: SrcPos, namePos: SrcPos, warnType: WarnTypes)
```

Reply:
I agree that it is simpler that way but here is why we might want to pass the whole tree rather than the pair of `SrcPos` (at least for imports). The thing is that with imports we have bunch of different cases:
1. single, regular import `import java.io.File`
2. multiple import `import java.io.{File, FileDescriptor}`
3. import with rename `import java.io.{File as JFile}
4. multiple imports with renames

In case of a multiple import there might be a case when only one of the selectors should be removed. This means going from: `import java.io.{File, FileDescriptor}` to `import java.io.{File}`. 

With the tree passed there one could do some better analysis in the method that will create a removal action (by e.g. pattern matching).